rust/src/proofs/api.rs

```diff
@@ -150,6 +150,24 @@ fn seal_pre_commit_phase1(
     })
 }
 
+/// TODO: document
+#[ffi_export]
+fn generate_sdr(
+    registered_proof: RegisteredSealProof,
+    cache_path: c_slice::Ref<u8>,
+    replica_id: &[u8; 32],
```

Comment:
Are you aware of any Go code utils for generating replica_id? (Or at least some docs / easy to understand rust code so that we can implement thin in Go)

Reply:
I wasn't sure if Lotus has that already, it seems it doesn't. It should be easy to implement. The code is at https://github.com/filecoin-project/rust-fil-proofs/blob/95e8c982540b14077d58ee9ddc076f7602b4805d/storage-proofs-porep/src/stacked/vanilla/params.rs#L758-L775. It really is just a normal SHA256 hash, nothing special.

If the inputs don't make sense (we have slightly different naming in our code base), let me know.